utils/error_injection.hh

```diff
@@ -454,6 +459,14 @@ public:
         co_await func(handler);
     }
 
+    future<> inject(const std::string_view& name, utils::wait_for_message wfm) {
```

Comment:
I think you are missing two customization properties:
* log-level (info, warning, etc)
* log/no-log 
If you can combine them, e.g. log-level:none then fine.
This PR adds more logs in places they weren't before and maybe are not required.
Adding the customization suggested here will enable keeping the code the same.

Reply:
Injected errors are only used by tests, there's no need in optimizing logging here